File: Chloe Gwon/2022-06-14-Find-Minimum-in-Rotated-Sorted-Array.js
var findMin = function(nums) {
    var left = 0;
    var right = nums.length-1;
    
    if (nums.length === 1){
        return nums[0];
    }
    
    while (left < right){
        var midPoint = left + Math.floor((right - left) / 2);
        
        if (nums[midPoint] > nums[right]){
            left = midPoint+1;
        } else if (nums[midPoint] < nums[right]){
            right = midPoint;
        }
    }
    return nums[left];
};

File: MinjiJu/2022-06-14-find-minimum-in-rotated-sorted-array.cpp
// binary search O(log(n))
class Solution {
public:
    int findMin(vector<int>& nums) {
        int lo=0, hi=nums.size()-1;
        
        while(lo<hi){
            auto mid = lo + (hi-lo)/2;
            
            if(nums[mid]<nums[hi]) hi = mid;
            else if(nums[mid]>nums[hi]) lo = mid+1;
        }
        return nums[lo];
    }
};


class Solution {
public:
    int findMin(vector<int>& nums) {
        
        int left=0, right=nums.size()-1;        

        while(left < right){
            
            if(nums[left] < nums[right]) {
                return nums[left];
            }
            
            int mid = (left+right)/2;
            
            if(nums[left] > nums[mid]){
                right = mid;
            } else {
                left = mid+1;
            }
        }
        return nums[left];
    }
};


class Solution {
public:
    int findMin(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        return nums[0];
    }
};